### PR TITLE
add tag key validation

### PR DESCRIPTION
**Pull request by rocktavious**

### CODE REVIEW

File: tags.go [hunk 1]
```diff
@@ -49,6 +59,15 @@ type TagDeleteInput struct {
 	Id ID `json:"id"`
 }
 
+//#region Helpers
+
+func ValidateTagKey(key string) error {
```
Comment (davidbloss) [inline tags.go:64]: What do you think about making this a `func (t *Tag) ValidateTagKey()...` method?
The `TagKeyRegex` and `TagKeyErrorMsg` vars could live in there too.

Comment (rocktavious) [reply tags.go:64]: ou i like that idea!

Comment (rocktavious) [reply tags.go:64]: this was not possible - we paired on it together

**Review by davidbloss — APPROVED** [review]
LGTM 🎉

**Pull request merged**